File: puppeteer-python/src/pages/home_page.py
"""Port of epigijon.devorapp.e2e.functional.pages.HomePage.

Page object for ``/home``. ``create()`` waits until the URL contains
``/home`` and the top navigation bar is visible before returning.
"""
from src.common.by import By
from src.pages.base_page import BasePage


class HomePage(BasePage):
    TOP_BAR = By.css(".topbar, header, nav")

    async def _wait_ready(self) -> None:
        await self.waiter.wait_for_home_page()
        await self.waiter.wait_for_top_bar()

    async def is_top_bar_visible(self) -> bool:
        return await self.is_visible(self.TOP_BAR)

    async def get_current_url(self) -> str:
        return self.page.url

File: playwright-csharp/Pages/HomePage.cs
using Microsoft.Playwright;

namespace DevorApp.E2ETests.Pages;

/// <summary>
/// Page object for /home. Port of
/// epigijon.devorapp.e2e.functional.pages.HomePage. <c>CreateAsync</c>
/// waits until the URL contains /home and the top navigation bar is
/// visible before returning.
/// </summary>
public class HomePage : BasePage
{
    private ILocator TopBar => Page.Locator(".topbar, header, nav");

    private HomePage(IPage page) : base(page)
    {
    }

    public static Task<HomePage> CreateAsync(IPage page) => InitAsync(new HomePage(page));

    protected override async Task WaitReadyAsync()
    {
        await Page.WaitForURLAsync(url => url.Contains("/home"));
        await TopBar.WaitForAsync();
    }

    public Task<bool> IsTopBarVisibleAsync() => TopBar.IsVisibleAsync();

    public string GetCurrentUrl() => Page.Url;
}
